src/x3f_image.c

```diff
@@ -166,16 +168,56 @@
   return 1;
 }
 
+/* extern */ int x3f_get_camf_rect(x3f_t *x3f, char *name,
+				   x3f_area16_t *image, int rescale,
+				   uint32_t *rect)
+{
+  if (!x3f_get_camf_matrix(x3f, name, 4, 0, 0, M_UINT, rect))
+    return 0;
+
+  return x3f_transform_rect_to_keep_image(x3f, image, rescale, rect);
+}
+
+/* extern */ int x3f_crop_area_column(x3f_t *x3f, char *name,
+				      x3f_area16_t *image, int rescale,
+				      x3f_area16_t *crop)
+{
+  uint32_t rect[4];
+  uint32_t column[4];
+
+  if (!x3f_get_camf_matrix(x3f, "DarkShieldColRange", 2, 2, 0, M_UINT, column))
+    return 0;
+
+  rect[1] = 0;           /* Cropped automatically later */
+  rect[3] = UINT32_MAX;  /* Cropped automatically later */
+
+  if (!strcmp(name, "FakeDarkShieldLeft")) {
```

Comment:
Yes, C is very primitive in this regard. I doesn't have atoms, and it doesn't really have sting manipulation either. All it has it sting constants. The rest is handled by th library.

I don't think indexes are really wrong here though. It's after all just a list of column rages. There is no guarantee that one of them will be to the left and the other to the right, or even that it will be two of them.